### PR TITLE
Bump configuration-as-code from 1.34 to 1.35

### PR DESCRIPTION
**Pull request by timja**
cc @casz

### CODE REVIEW

**Comment by timja** [comment]
Currently failing on:
https://github.com/jenkinsci/script-security-plugin/pull/284
https://github.com/jenkinsci/git-client-plugin/pull/490

**Comment by casz** [comment]
let's remove the fail fast? 🤔

**Comment by timja** [comment]
> let's remove the fail fast, while testing? 🤔

if i change the Jenkinsfile it will just be ignored, 
maybe @jglick will change it for us 😉

**Comment by casz** [comment]
if we ask nicely maybe we can get commit access? 🙇

**Comment by timja** [comment]
> if we ask nicely maybe we can get commit access? 🙇

Pretty please Jesse 😄

**Comment by timja** [comment]
Seeing if the ansicolor failure is a flake or consistent

**Comment by timja** [comment]
failed this time on script-security. 
would be good if we could get failFast disabled so we can see all the errors though.

**Comment by timja** [comment]
FTR https://github.com/jenkinsci/script-security-plugin/pull/284#issuecomment-578290120

**Comment by jglick** [comment]
> would be good if we could get failFast disabled

I am reluctant to disable `failFast` in `master`. Builds are very expensive, and there are a lot of failing PRs. I suppose I can try to load this flag from a separate file, so untrusted contributors can set it temporarily in a PR. Also I can give some people commit access.

**Comment by casz** [comment]
Seems we still have some work ahead of us 😅

**Comment by timja** [comment]
Thanks Jesse,
Currently failing:

ansicolor (test failures)
cloudbees-folder (test failures)
command-launcher (test failures, looks like a flake / infra issue)
git-client (test failures)
matrix-project (test failures)
plain-credentials (test failures)

Seems like none are failing for jcasc test-harness anymore.

I'm going to give it a kick to see if they stay consistent

**Comment by timja** [comment]
Failures were the same, needs investigation.

**Comment by timja** [comment]
Had a look at:
git-client: Issue with tests relying on external state https://github.com/jenkinsci/git-client-plugin/pull/496/files/d6beaef005cb459bbcadbc18a8d0eeaf391c1906..c9e2dfa8dd43f31e0deb9de2c412de5fe392d6c9
command-launcher: from what I can tell it's picking up the detached plugin instead of the latest version, @jglick if you happen to have a chance would be great to get your thoughts on it, I came across this in my hunt: https://github.com/jenkinsci/jenkins/pull/4100

**Comment by jglick** [comment]
> command-launcher: from what I can tell it's picking up the detached plugin instead of the latest version

The fix was in 2.185 AFAICT so maybe just suppress the failure on older lines?

**Comment by timja** [comment]
> > command-launcher: from what I can tell it's picking up the detached plugin instead of the latest version
> 
> The fix was in 2.185 AFAICT so maybe just suppress the failure on older lines?

Ugh it was already being ignored, but my patch to make jcasc failures fail caused that to be ignored, let me revert and see how we're looking

File: pct.sh [hunk 1]
```diff
@@ -29,9 +29,10 @@ java -jar pct.jar \
      -mavenProperties "$MAVEN_PROPERTIES" \
      -skipTestCache true
 
-if fgrep -q '<status>INTERNAL_ERROR</status>' pct-report.xml
+if grep -q -F -e '<status>INTERNAL_ERROR</status>' -e '<status>TEST_FAILURES</status>' pct-report.xml
```
Comment (timja) [inline pct.sh]:
```suggestion
if grep -q -F -e '<status>INTERNAL_ERROR</status>' pct-report.xml
```

**Comment by casz** [comment]
ssh-credentials fails because of `Credentials Plugin v2.1.11 is older than required. To fix, install v2.2.0 or later.` 🤔

**Comment by timja** [comment]
> ssh-credentials fails because of `Credentials Plugin v2.1.11 is older than required. To fix, install v2.2.0 or later.` 🤔

I saw that earlier, but didn't understand it as the plugin doesn't depend on it, there must be a transitive dep i guess

**Comment by timja** [comment]
Created https://github.com/jenkinsci/cloudbees-folder-plugin/pull/142 anyway, should fix it

**Comment by timja** [comment]
Haven't managed to figure this out yet, if anyone has ideas:
https://github.com/jenkinsci/workflow-step-api-plugin/blob/051fb8ed1a65a9fa9040a024b7a5bae3c57ccd97/pom.xml#L88
workflow-step-api is using 1.20 and bom is using 1.20
```
Error
Plugin workflow-step-api failed to start
Stacktrace
java.lang.Error: Plugin workflow-step-api failed to start
	at org.jvnet.hudson.test.PluginAutomaticTestBuilder$OtherTests.testPluginActive(PluginAutomaticTestBuilder.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.jvnet.hudson.test.junit.GroupedTest.runGroupedTests(GroupedTest.java:67)
	at org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite.doTests(JellyTestSuiteBuilder.java:152)
	at org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite.access$100(JellyTestSuiteBuilder.java:138)
	at org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite$2.call(JellyTestSuiteBuilder.java:160)
	at org.jvnet.hudson.test.HudsonTestCase$WebClient$5.run(HudsonTestCase.java:1514)
	at org.jvnet.hudson.test.ClosureExecuterAction.doIndex(ClosureExecuterAction.java:53)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:396)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:408)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:212)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:145)
	at org.kohsuke.stapler.IndexDispatcher.dispatch(IndexDispatcher.java:27)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:878)
	at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:456)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:878)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:676)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:154)
	at jenkins.telemetry.impl.UserLanguages$AcceptLanguageFilter.doFilter(UserLanguages.java:128)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:157)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:105)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:564)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
	at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:126)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Pipeline: Step API version 2.22 failed to load.
 - Structs Plugin version 1.7 is older than required. To fix, install version 1.20 or later.
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:921)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:544)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1095)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	... 3 more
Standard Output
```

**Comment by timja** [comment]
I can re-produce locally with:
```
mvn --show-version --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --define=failIfNoTests=false --define=jth.jenkins-war.path=/Users/timja/projects/jenkinsci/bom/target/local-test/megawar.war --define=org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --define=forkCount=1 hpi:resolve-test-dependencies hpi:test-hpl surefire:test -Djenkins.version=2.176.3 -Denforcer.skip=true
```

Without the megawar is passes fine

If I unzip the megawar the jenkins.version is correct and structs is 1.20

any ideas 😄

**Comment by casz** [comment]
Some other plugin that pulls structs plugin down?

**Comment by timja** [comment]
The oldest I've found is scm-api has 1.9, but the error message is talking about 1.7,
I unzipped and grepped for structs =/

**Comment by amuniz** [comment]
I think the problem with structs comes from the fact that it is a detached plugin, so if some installed plugin depends on it but its version is not specified, then the detached one (usually an old one) is installed.

[`workflow-step-api` depends on `structs`](https://github.com/jenkinsci/workflow-step-api-plugin/blob/eb1ecf23fe7ec9f5ad03e53c3a5ca2b2d913a985/pom.xml#L90), which version should [come from `bom-2.138.x` version `4`](https://github.com/jenkinsci/workflow-step-api-plugin/blob/eb1ecf23fe7ec9f5ad03e53c3a5ca2b2d913a985/pom.xml#L74-L76), however there is no version for `structs` in [that bom release](https://github.com/jenkinsci/bom/blob/bom-4/bom-2.138.x/pom.xml).

So, I think a fix is needed in `workflow-step-api`, either to upgrade the `bom` dependency to one including `structs` or to set the version of the `structs` dependency directly.

**Comment by amuniz** [comment]
Oh, and this is why I don't like the PCT... it's testing a quite unreal setup.

**Comment by timja** [comment]
thanks for taking a look @amuniz,

From what I can see it should be managed though,
https://github.com/jenkinsci/bom/blob/bom-4/bom-latest/pom.xml#L249

In that bom release it's importing other boms up until latest, 
https://github.com/jenkinsci/bom/blob/bom-4/bom-2.138.x/pom.xml#L13-L19

**Comment by amuniz** [comment]
How is bom-latest being picked up? I don't see it in https://github.com/jenkinsci/bom/blob/bom-4/bom-2.138.x/pom.xml

It looks they are picked up backward, but not forward (which seems correct).

**Comment by timja** [comment]
If I run:
```
mvn help:effective-pom
```

I see:
```
      <dependency>
        <groupId>org.jenkins-ci.plugins</groupId>
        <artifactId>structs</artifactId>
        <version>1.20</version>
      </dependency>

```

https://github.com/jenkinsci/bom/blob/bom-4/bom-2.138.x/pom.xml#L15

https://github.com/jenkinsci/bom/blob/bom-4/bom-2.150.x/pom.xml#L15

...

**Comment by amuniz** [comment]
Ah, ok. Then I have no idea :man_shrugging:

**Comment by timja** [comment]
@jglick any ideas?

**Comment by timja** [comment]
Also @amuniz the bom change in workflow-step-api is unreleased, but even if I checkout the released version I have the same issue,

https://github.com/jenkinsci/workflow-step-api-plugin/pull/54/files#diff-600376dffeb79835ede4a0b285078036L88

**Comment by jglick** [comment]
Sorry, I have no time to work on this at the moment.

> this is why I don't like the PCT... it's testing a quite unreal setup

Yes, no one likes it really, but we are sort of stuck with it because nearly all our test coverage presumes a Maven flat `test` classpath. There have been abortive efforts to create better systems.

File: failFast [hunk 2]
```diff
@@ -1 +1 @@
-true
+false
```
Comment (timja) [inline failFast]: true

**Comment by jglick** [comment]
You are busy! Thanks for moving this repo along.

**Pull request merged**